### PR TITLE
upgrade to latest xDSL (stacked 1/3)

### PR DESCRIPTION
**Pull request by AntonLydike**
Tiny PR to update to latest xDSL

### CODE REVIEW

**Comment by AntonLydike** [comment]
This update in xdsl fixed some variable name reuse issues, so I had to update a bunch of tests, but bundling this change with no changes to the rewrites seems to be the best way to assure everyone that no functional differences linger within the test diffs.

**Comment by AntonLydike** [comment]
halp, docker broke! :sob:

**Comment by jorendumoulin** [comment]
The snax docker will become deprecated hopefully very soon, so for me we can ignore the failing test.

File: requirements.txt [hunk 1]
```diff
@@ -2,5 +2,5 @@ pre-commit
 filecheck
 lit
 numpy
-xdsl @ git+https://github.com/xdslproject/xdsl.git@5bd448e56719508b9f3149167430f0a8b7c66574
+xdsl @ git+https://github.com/xdslproject/xdsl.git@2d2e3e5f274210e8f19b03fc98c4033fd0bc3ee4
```
Comment (jorendumoulin) [inline requirements.txt:5]:
```suggestion
xdsl @ git+https://github.com/xdslproject/xdsl.git@168758fee7fda1beec84bab81b249542d76fa188
```

File: compiler/tools/snax_opt_main.py [hunk 2]
```diff
@@ -45,6 +45,9 @@ def __init__(
         super().register_all_targets()
 
         ## Add custom dialects & passes
+        # TODO: this is needed with newer xDSL as they now also have the dialect
```
Comment (JosseVanDelm) [inline compiler/tools/snax_opt_main.py:48]: nit:
```suggestion
        # FIXME: override upstream accfg dialect. Remove this after upstreaming full downstream accfg dialect.  
```

File: requirements.txt (hunk 1 above)
Comment (jorendumoulin) [reply requirements.txt:5]: Let's use the most recent version as we just merged a nice fix for something else :)

**Review by jorendumoulin — APPROVED** [review]

**Review by JosseVanDelm — APPROVED** [review]
:sunglasses:

**Pull request merged**